schema/asyncapi.json

```diff
@@ -1,6 +1,6 @@
 {
-  "title": "AsyncAPI 1.2.0 schema.",
-  "id": "http://asyncapi.hitchhq.com/v1/schema.json#",
+  "title": "AsyncAPI 2.0.0 schema.",
+  "id": "http://asyncapi.hitchhq.com/v2/schema.json#",
```

Comment:
Let's not raise the version here. I'll do it when releasing a new version. BTW, is there any breaking change in this PR?

Reply:
This should be non-breaking as I've only added new objects Actually thought of another useful change to the schema on the OAuth flow objects so I've bring that in as well when I correct the version number

Reply:
Changes added as request - please see updates in pull request